examples/bri-3/src/bri/workgroup/worksteps/agents/workstepsStorage.agent.ts

```diff
@@ -41,35 +43,26 @@ export class WorkstepStorageAgent extends PrismaService {
       },
     });
     return workstepModels.map((w) => {
-      return new Workstep(
-        w.id,
-        w.name,
-        w.version,
-        w.status,
-        w.workgroupId,
-        w.securityPolicy,
-        w.privacyPolicy,
-      );
+      return this.mapper.map(w, Workstep, Workstep) as Workstep;
     });
   }
 
   async createNewWorkstep(workstep: Workstep): Promise<Workstep> {
     const newWorkstepModel = await this.workstep.create({
-      data: this.mapper.map(workstep, getType<Workstep>()) as Workstep,
+      data: workstep,
     });
-    return this.mapper.map(newWorkstepModel, getType<Workstep>()) as Workstep;
+    return this.mapper.map(newWorkstepModel, Workstep, Workstep) as Workstep;
```

Comment:
the two worksteps are the old and the new, right? @Manik-Jain is there a way to distinguish both?

Reply:
mapper accepts source and destination...
when we see two same names that's when we are converting from prisma domain type to our domain type.